SC2148 on foo.bash without shebang

Contrary to the command-line behavior of `shellcheck`, I get an error on a simple include-only bash "module" named `foo.bash` that doesn't have a shebang, like:
```bash
function do_foo() {
  echo "hello world"
}
```

running shellcheck from the command line yields:
```shell
$ shellcheck foo.bash
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.6.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
$
```

According to [the SC2148 wiki](https://github.com/koalaman/shellcheck/wiki/SC2148), it should be fine with a `.bash` extension. It does drop the error if I add the alternative `# shellcheck shell=ksh`, but it seems that shouldn't be necessary.

--- a/.vscode/tasks.json
+++ b/.vscode/tasks.json
@@ -16,6 +16,7 @@
             }
         },
         {
+            "label": "build",
             "type": "typescript",
             "tsconfig": "tsconfig.json",
             "problemMatcher": [

--- a/src/linter.ts
+++ b/src/linter.ts
@@ -253,7 +253,7 @@ export default class ShellCheckProvider {
 
             const executable = settings.executable || 'shellcheck';
             const diagnostics: vscode.Diagnostic[] = [];
-            let processShellCheckItem = (item: ShellCheckItem) => {
+            const processShellCheckItem = (item: ShellCheckItem) => {
                 if (item) {
                     diagnostics.push(makeDiagnostic(textDocument, item));
                 }
@@ -262,6 +262,14 @@ export default class ShellCheckProvider {
             let args = ['-f', 'json'];
             if (settings.exclude.length) {
                 args = args.concat(['-e', settings.exclude.join(',')]);
+            }
+
+            // https://github.com/timonwong/vscode-shellcheck/issues/43
+            // We should explicit set shellname based on file extension name
+            const fileExt = path.extname(textDocument.fileName);
+            if (fileExt === '.bash' || fileExt === '.ksh' || fileExt === '.dash') {
+                // shellcheck args: specify dialect (sh, bash, dash, ksh)
+                args = args.concat(['-s', fileExt.substr(1)]);
             }
 
             if (settings.customArgs.length) {

